docs/applications/draco_learn.ipynb

```diff
@@ -713,9 +634,128 @@
    ],
    "source": [
     "features = data.negative.columns\n",
+    "new_weights = {}\n",
     "\n",
     "for feature, weight in zip(features, clf.coef_[0]):\n",
-    "    print(f\"#const {feature}_weight = {int(weight * 1000)}.\\n\")"
+    "    print(f\"#const {feature}_weight = {int(weight * 1000)}.\\n\")\n",
+    "    new_weights[f\"{feature}_weight\"] = int(weight * 1000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0ce3b548",
+   "metadata": {},
+   "source": [
+    "## How to use the learned weights?\n",
+    "\n",
+    "For how to do recomendation with default weights, refer to [Visualization Recomendation](https://dig.cmu.edu/draco2/applications/recommendation.html). \n",
+    "\n",
+    "In this example, we use the learned weights above to recommand visualizations.\n",
+    "\n",
+    "First, we pass the new learned weights to Draco as following:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "d970495a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from draco import answer_set_to_dict\n",
+    "from pprint import pprint\n",
+    "\n",
+    "new_draco = Draco(weights=new_weights)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c9705fad",
+   "metadata": {},
+   "source": [
+    "Then, we can continue with the same procedure as in default [Visualization Recommendation](https://dig.cmu.edu/draco2/applications/recommendation.html). \n",
+    "And as you can see, the recommended result from the same input based on the learned weights is different from the one in default. \n",
+    "\n",
+    "Note that whether it generates better recommendations or not depends on the examples you learned from. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "b7075ffd",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "INPUT:\n",
+      "attribute(number_rows,root,100).\n",
+      "\n",
+      "entity(field,root,(f,0)).\n",
+      "attribute((field,name),(f,0),temperature).\n",
+      "attribute((field,type),(f,0),number).\n",
+      "attribute((field,unique),(f,0),100).\n",
+      "\n",
+      "entity(view,root,(v,0)).\n",
+      "\n",
+      "entity(mark,(v,0),(m,0)).\n",
+      "entity(encoding,(m,0),(e,0)).\n",
+      "attribute((encoding,field),(e,0),(f,0)).\n",
+      "attribute((encoding,binning),(e,0),10).\n",
+      "\n",
+      "#show entity/3.\n",
+      "#show attribute/3.\n",
+      "\n",
+      "OUTPUT:\n",
+      "{'field': [{'name': 'temperature', 'type': 'number', 'unique': 100}],\n",
+      " 'number_rows': 100,\n",
+      " 'task': 'summary',\n",
+      " 'view': [{'coordinates': 'polar',\n",
+      "           'mark': [{'encoding': [{'binning': 10,\n",
+      "                                   'channel': 'x',\n",
+      "                                   'field': ('f', 0)},\n",
```

Comment:
I would not expect to see `'field': ('f', 0)` here, `'field': 'temperature'` would make more sense in this context. Is  [histogram.lp](https://github.com/cmudig/draco2/blob/635085365cd5d0e23b8267424b0dec5ab998eead/draco/asp/examples/histogram.lp) in the correct format?

Reply:
As I remember, we had to use the `(x,0)` format in the generator to generate certain number of encodings etc. I can't remember if that's required for fields, though. It's refer to the entity id as `temperature` is for the name.

Reply:
Oh, do you actually mean the output dict format?

Reply:
Exactly, I was referring to the final output dict. In its current form it cannot be rendered. If this spec is valid, then I should adjust the renderer to handle it, otherwise we should explore why an invalid spec is produced.

Reply:
Thanks! We will address this in a separate pr for issue #467

Reply:
Perfect, thanks a lot!